### PR TITLE
fix: console error on click of connect online with plaid button

### PR DESCRIPTION
**Pull request by payal-lathidadiya**
### Fixed Issues
$ https://github.com/Expensify/App/issues/22448
PROPOSAL: https://github.com/Expensify/App/pull/21778#discussion_r1256482993

### Tests

1. Create workspace if not created
2. Click on Workspaces in setting in left panel
3. Choose Reimbursement
4. Click the Connect bank account button
5. Click the Connect online with Plaid
6. Verify that it redirect to plaid portal and no error is displayed in console.

- [X] Verify that no errors appear in the JS console

### QA Steps
Same as above

- [X] Verify that no errors appear in the JS console

### PR Author Checklist

- [X] I linked the correct issue in the `### Fixed Issues` section above
- [X] I wrote clear testing steps that cover the changes made in this PR
    - [X] I added steps for local testing in the `Tests` section
    - [X] I added steps for the expected offline behavior in the `Offline steps` section
    - [X] I added steps for Staging and/or Production testing in the `QA steps` section
    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I ran the tests on **all platforms** & verified they passed on:
    - [X] Android / native
    - [X] Android / Chrome
    - [X] iOS / native
    - [X] iOS / Safari
    - [X] MacOS / Chrome / Safari
    - [X] MacOS / Desktop
- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] I verified that if a function's arguments changed that all usages have also been updated correctly
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

image

</details>

<details>
<summary>Mobile Web - Chrome</summary>

image

</details>

<details>
<summary>Mobile Web - Safari</summary>

image

</details>

<details>
<summary>Desktop</summary>

image

</details>

<details>
<summary>iOS</summary>

image

</details>

<details>
<summary>Android</summary>

image

</details>

### CODE REVIEW

File: src/pages/ReimbursementAccount/BankAccountPlaidStep.js
Comment (Santhosh-Sellavel) [inline src/pages/ReimbursementAccount/BankAccountPlaidStep.js:137]: Let's remove the `useLocalize` hook and use the `translate` from the `props`.

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
Otherwise Looks good, tests well!

**Comment by Santhosh-Sellavel** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](httpxs://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/85645967/104680e6-9f97-41ef-a823-e0b5ebaa8ae4

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/85645967/23cd3e7a-dd99-466e-80de-dfa865b5638f

</details>

<details>
<summary>iOS & Android</summary>

https://github.com/Expensify/App/assets/85645967/5e3d0906-1697-422e-83b0-355ac89700fc

</details>

File: src/pages/ReimbursementAccount/BankAccountPlaidStep.js
Comment (payal-lathidadiya) [reply src/pages/ReimbursementAccount/BankAccountPlaidStep.js:135]: Makes sense. updated

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM, all yours @francoisl!

**Review by hayata-suenaga — DISMISSED** [review]
Simple PR to remove `useLocalize` and bring back `withLocalize` LGTM 👍 

just a question, why we cannot `useLocalize`?

@francoisl I'll wait until you have a chance to review since you were assigned to the migration PR that caused the regression

**Comment by Santhosh-Sellavel** [comment]
> just a question, why we cannot useLocalize?

We can do that too, but we need to remove `withLocalizeProptypes` from `StepPropTypes` which is common to multiple pages.  Maybe could have just omitted `withLocalizeProptypes` from `StepPropTypes` this component alone.

**Comment by Santhosh-Sellavel** [comment]
> Maybe could have just omitted withLocalizeProptypes from StepPropTypes for this component alone.

@payal-lathidadiya Can you just do that, Sorry for the back and forth!

**Comment by francoisl** [comment]
Yes, as far as I know, using the `useLocalize` hook is the preferred way moving forward. If possible, let's keep using that and the fix the issue with the extra prop types.

**Review by Santhosh-Sellavel — APPROVED** [review]
All yours @francoisl!

**Review by Santhosh-Sellavel — APPROVED** [review]

**Comment by payal-lathidadiya** [comment]
> > Maybe could have just omitted withLocalizeProptypes from StepPropTypes for this component alone.
> 
> @payal-lathidadiya Can you just do that, Sorry for the back and forth!

np. updated and pushed.

**Comment by payal-lathidadiya** [comment]
@Santhosh-Sellavel Post omitting withLocalizeProptypes from StepPropTypes, somehow ESLint is failing for rule onyx-props-must-have-default
image

Not sure what's cause of this. any idea?

**Review by hayata-suenaga — COMMENTED** [review]
do we have to do `..._.omit(StepPropTypes, _.keys(withLocalizePropTypes)),` in components that use `useLocalize` and StepProps?

**Comment by payal-lathidadiya** [comment]
> do we have to do `..._.omit(StepPropTypes, _.keys(withLocalizePropTypes)),` in components that use `useLocalize` and StepProps?

Yeah, since `StepProps` contains `withLocalizePropTypes` which is from HOC and If we replace use of `withLocalize` HOC with `useLocalize` hook, we'll have to omit those props injected from that HOC.

Else, it will result in console below errors (ref https://github.com/Expensify/App/issues/22448)
image

**Comment by hayata-suenaga** [comment]
Should we remove `withLocalizePropTypes` from `StepProps` down the line once all components are migrated to function components?

**Comment by hayata-suenaga** [comment]
okay thanks for explaining

I re-ran the lint job again by the way. it failed for some unknow reason

**Comment by payal-lathidadiya** [comment]
@hayata-suenaga @Santhosh-Sellavel 
To unblock us, I'm planning to revert this fix to previous where we'll remove `useLocalize` hook for now and add back `withLocalize` HOC.
Let me know if you think otherwise.

**Comment by Santhosh-Sellavel** [comment]
@francoisl On the other hand let's just do nothing here as it will be eventually removed. Thoughts

**Comment by francoisl** [comment]
That could be an option. Are those 3 refactors in progress already or still on hold? Also, to clarify, other that the console error, is this causing any crash or breaking anything at the moment?

**Comment by payal-lathidadiya** [comment]
> That could be an option. Are those 3 refactors in progress already or still on hold? Also, to clarify, other that the console error, is this causing any crash or breaking anything at the moment?

@francoisl  for those components, migration and refactor is still on hold.

Below is state of component which uses StepPropTypes and withLocalize HOC
1. ACHContractStep - migrated to function component 
2. BankAccountStep - migrated to function component 
3. BankAccountManualStep -  migrated to function component 
4. BankAccountPlaidStep - migrated to function component [current migration]
5. **CompanyStep - class component, yet to migrate [on Hold]**
6. **RequestorOnfidoStep - class component, yet to migrate [on Hold]**
7. **RequestorStep - class component, yet to migrate [on Hold]**

Also, this isn't causing any crash or breaking anything at the moment.

**Comment by francoisl** [comment]
Ok thanks. If some of those refactors are still on hold without a clear estimate of when they'll be done, I'd say it's fine to keep using the `withLocalize` HOC for now, but I'll defer to @hayata-suenaga for the final decision.

**Comment by hayata-suenaga** [comment]
let's use `withLocalize` for now 👍

**Review by hayata-suenaga — APPROVED** [review]

**Comment by payal-lathidadiya** [comment]
Once all components which uses `StepPropTypes` are migrated to function component, we can remove withLocalizePropTypes from `StepPropTypes` down the line.

And remove withLocalize HOC reference and replace use of `translate` from `useTranslate` hook instead of props.

@francoisl / @hayata-suenaga should we create a new backlog issue for this expected refactor requirement in migration epic.

**Comment by payal-lathidadiya** [comment]
I think we already have got approvals. 
If everything is good, can this be merged pls.

**Comment by hayata-suenaga** [comment]
@allroundexperts please final approve this PR when you got a moment 🙇

**Review by allroundexperts — APPROVED** [review]
Looks good!

**Review by Santhosh-Sellavel — APPROVED** [review]
bump @hayata-suenaga!

**Review by hayata-suenaga — APPROVED** [review]
👍

**Comment by kavimuru** [comment]
@payal-lathidadiya We get console error in this step `Verify that it redirect to plaid portal and no error is displayed in console.`
Do we need separate GH ticket for this?

**Comment by Santhosh-Sellavel** [comment]
@kavimuru please share the console error please.

**Comment by kavimuru** [comment]
@Santhosh-Sellavel this is the video

https://github.com/Expensify/App/assets/43996225/6edb17d4-d7ac-470f-98a0-6c6285e08453

**Comment by payal-lathidadiya** [comment]
@kavimuru 
This seems to be a different issue and not related to current fix.

I just checked, this exists in production as well;

https://github.com/Expensify/App/assets/130229379/135ab24b-48b7-44cb-8371-4e906b06e692

**Comment by Santhosh-Sellavel** [comment]
@kavimuru please a new issue that's not relevant here.

**Pull request merged**
